internal/invocation/stripe_executor.go

```diff
@@ -0,0 +1,97 @@
+package invocation
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"sync"
+)
+
+var (
+	// Default values differ from java impl. Also queue size is calculated differently.
+	// Java Client: queueSize per worker = defaultEventQueueCapacity / defaultEventWorkerCount
+	// Go Client: queueSize per worker = defaultEventQueueCapacity
+	defaultEventQueueCapacity = 10000
+	defaultEventWorkerCount   = runtime.NumCPU()
+)
+
+// executor represents the function that will run on workers of stripeExecutor.
+type executor func(queue chan func(), quit chan struct{}, wg *sync.WaitGroup)
+
+// stripeExecutor executes given "tasks" preserving the order among the ones that are given with the same key.
+type stripeExecutor struct {
+	quit       chan struct{}
+	wg         *sync.WaitGroup
+	execFn     executor
+	taskQueues []chan func()
+	queueCount int
+}
+
+// newStripeExecutor returns a new stripeExecutor with default configuration.
+func newStripeExecutor() stripeExecutor {
+	// ignore error, default values do not raise error.
+	ex, _ := newStripeExecutorWithConfig(defaultEventWorkerCount, defaultEventQueueCapacity)
+	return ex
+}
+
+// newStripeExecutor returns a new stripeExecutor with configured queueCount and queueSize.
+func newStripeExecutorWithConfig(queueCount, queueSize int) (stripeExecutor, error) {
+	if queueCount <= 0 {
+		return stripeExecutor{}, fmt.Errorf("queueCount must be greater than 0")
+	}
+	if queueSize <= 0 {
+		return stripeExecutor{}, fmt.Errorf("queueSize must be greater than 0")
+	}
+	se := stripeExecutor{
+		taskQueues: make([]chan func(), queueCount),
+		queueCount: queueCount,
+	}
+	for i := range se.taskQueues {
+		se.taskQueues[i] = make(chan func(), queueSize)
+	}
+	se.quit = make(chan struct{})
+	se.wg = &sync.WaitGroup{}
+	se.execFn = defaultExecFn
+	return se, nil
+}
+
+// start fires up the workers for each queue.
+func (se stripeExecutor) start() {
+	se.wg.Add(int(se.queueCount))
+	for i := range se.taskQueues {
+		go se.execFn(se.taskQueues[i], se.quit, se.wg)
+	}
+}
+
+// dispatch sends the handler "task" to one of the appropriate taskQueues, "tasks" with the same key end up on the same queue. Returns true if queue is full and could not dispatch
+func (se stripeExecutor) dispatch(key int, task func()) (queueFull bool) {
```

Comment:
I think this function should return `false` when the task cannot be dispatched. That could also help getting rid of the return variable name.

Reply:
It is definitely more idiomatic thanks, addressed it at https://github.com/hazelcast/hazelcast-go-client/pull/696/commits/3bb2d0ff640966a075cfd6b125e5cf03c81abdc6